prow/pjutil/pjutil.go

```diff
@@ -144,23 +144,16 @@ func ProwJobToPod(pj kube.ProwJob, buildID string) (*v1.Pod, error) {
 		return nil, err
 	}
 
-	spec := pj.Spec.PodSpec
+	spec := pj.Spec.PodSpec.DeepCopy()
 	spec.RestartPolicy = "Never"
 
-	// Set environment variables in each container in the pod spec. We don't
-	// want to update the spec in place, since that will update the ProwJob
-	// spec. Instead, create a copy.
-	spec.InitContainers = []v1.Container{}
```

Comment:
Won't this removal result in doubling the env variables inside the containers?

Reply:
I don't think so? Before we started with an empty slice and copied each container over one by one, adding env to it as we went. Now we DeepCopy the slice and add env to each in place.

Reply:
We can't DeepCopy _and_ do this assignment, or we copy the slice, then clear it with this assignment, then try to in-place edit an empty slice?

Reply:
Also the unit tests would have caught a mismatch on env